web/src/drafts_overlay_ui.ts

```diff
@@ -269,25 +292,28 @@ function setup_event_handlers(): void {
 }
 
 export function launch(): void {
-    const all_drafts = drafts.draft_model.get();
-    const narrow_drafts = drafts.filter_drafts_by_compose_box_and_recipient(all_drafts);
-    const other_drafts = _.pick(
-        all_drafts,
-        _.difference(Object.keys(all_drafts), Object.keys(narrow_drafts)),
-    );
-    const formatted_narrow_drafts = format_drafts(narrow_drafts);
-    const formatted_other_drafts = format_drafts(other_drafts);
-
-    render_widgets(formatted_narrow_drafts, formatted_other_drafts);
+    const {narrow_drafts, other_drafts, narrow_drafts_header} = get_formatted_drafts_data();
```

Comment:
Moving this into `get_formatted_drafts_data` can be its own prep commit

Reply:
Then you can have another prep commit for caching the header in `get_formatted_drafts_data` return value instead of calculating it each time.